### PR TITLE
Add @PreDetroy annotation in services implementing Purgeable

### PR DESCRIPTION
**Pull request by nabil-Tounarti**

### CODE REVIEW

**Review by Natchica — APPROVED** [review]

File: src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java [hunk 1]
```diff
@@ -531,12 +532,13 @@ private TaskAbortCause getTaskAbortCause(Task task) {
 
     // region purge locks
     @Override
-    public boolean purgeTask(String chainTaskId) {
+    public boolean purgeTask(final String chainTaskId) {
```
Comment (Natchica) [inline src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java:535]:
```suggestion
    public boolean purgeTask(final String chainTaskId) {
        log.debug("purgeTask [chainTaskId: {}]", chainTaskId);
```

File: src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java [hunk 2]
```diff
@@ -531,12 +532,13 @@ private TaskAbortCause getTaskAbortCause(Task task) {
...
         taskAccessForNewReplicateLocks.remove(chainTaskId);
         return !taskAccessForNewReplicateLocks.containsKey(chainTaskId);
     }
 
     @Override
+    @PreDestroy
     public void purgeAllTasksData() {
         taskAccessForNewReplicateLocks.clear();
```
Comment (Natchica) [inline src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java:543]:
```suggestion
        log.info("Method purgeAllTasksData() called to perform task data cleanup.");
        taskAccessForNewReplicateLocks.clear();
```

**Review by Natchica — APPROVED** [review]

File: src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java (hunk 1 above)
Comment (nabil-Tounarti) [reply src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java:535]: resolved in : [a35058a](https://github.com/iExecBlockchainComputing/iexec-core/pull/724/commits/a35058a652778f31cfebf678aab460f24ca556b8)

File: src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java (hunk 2 above)
Comment (nabil-Tounarti) [reply src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java:545]: resolved in : [a35058a](https://github.com/iExecBlockchainComputing/iexec-core/pull/724/commits/a35058a652778f31cfebf678aab460f24ca556b8)

**Review by jbern0rd — APPROVED** [review]
LGTM !

**Pull request merged**